src/states/node.rs

```diff
@@ -636,12 +636,43 @@ impl Node {
             (ExpectCloseNode { expect_id, client_auth, message_id }, Section(_), Section(_)) => {
                 self.handle_expect_close_node_request(expect_id, client_auth, message_id)
             }
-            (ConnectionInfo(conn_info), src @ Client { .. }, dst @ ManagedNode(_)) => {
-                self.handle_connection_info_from_client(conn_info, src, dst)
-            }
-            (ConnectionInfo(conn_info), ManagedNode(src_name), dst @ Client { .. }) |
-            (ConnectionInfo(conn_info), ManagedNode(src_name), dst @ ManagedNode(_)) => {
-                self.handle_connection_info_from_node(conn_info, src_name, dst)
+            (ConnectionInfoRequest { encrypted_connection_info,
+                                     nonce_bytes,
+                                     public_id,
+                                     message_id },
+             src @ Client { .. },
+             dst @ ManagedNode(_)) |
+            (ConnectionInfoRequest { encrypted_connection_info,
+                                     nonce_bytes,
+                                     public_id,
+                                     message_id },
+             src @ ManagedNode(_),
+             dst @ ManagedNode(_)) => {
+                self.handle_connection_info_request(encrypted_connection_info,
+                                                    nonce_bytes,
+                                                    public_id,
+                                                    message_id,
+                                                    src,
+                                                    dst)
+            }
+            (ConnectionInfoResponse { encrypted_connection_info,
+                                      nonce_bytes,
+                                      public_id,
+                                      message_id },
+             ManagedNode(src_name),
+             dst @ Client { .. }) |
+            (ConnectionInfoResponse { encrypted_connection_info,
+                                      nonce_bytes,
+                                      public_id,
+                                      message_id },
+             ManagedNode(src_name),
+             dst @ ManagedNode(_)) => {
+                self.handle_connection_info_response(encrypted_connection_info,
+                                                     nonce_bytes,
+                                                     public_id,
+                                                     message_id,
+                                                     src_name,
+                                                     dst)
```

Comment:
Not important, but I wonder whether with names like `enc_conn_info`, `pub_id`, `msg_id` and `nonce` we could save some vertical space here and make this a bit easier on the eye.